src/Docker.DotNet/MultiplexedStreamReader.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Docker.DotNet
+{
+    public class MultiplexedStreamReader
+    {
+        private readonly MultiplexedStream _stream;
+
+        public MultiplexedStreamReader(MultiplexedStream stream)
+        {
+            _stream = stream;
+        }
+
+        public async Task<string> ReadLineAsync(CancellationToken token = default)
+        {
+            var line = new List<byte>();
+
+            var buffer = new byte[1];
+
```

Comment:
@The-TT-Hacker I tested this method trying to get rid of headers being shown on `MonitorStreamAsync` method and haven't managed to cancel it. 
Would you confirm if expected behaviour is to read until 0 bytes are returned or to stop reading when no more lines are retrieved?
When `res` equals 0, it loops indefinitelly.
This is test I used
https://gist.github.com/dgvives/2620960049191797ea0d871c339d9bc2#file-program-cs-L1 

It would be nice if MultiplexedStreamReader could detect stream being multiplexed  and provide output por ReadLineAsync without control chars.

Have you checked this running containers on different platforms? Because of EOL

Reply:
@ACoderLife hey! Sorry for taking long. I'm cleaning up and merging so PRs with fixes and improvements before I start the wave of changes on the whole codebase. 

That being said, have you checked @dgvives concern with the infinite loop?

Thank you!